### PR TITLE
Enable modification of ELB attributes such as draining and timeouts

### PR DESCRIPTION
**Pull request by jvasallo**
Adding support for users to add elb idle connection timeouts as well as specify connection draining.

### CODE REVIEW

File: src/foremast/elb/create_elb.py [hunk 1]
```diff
@@ -208,3 +221,68 @@ def add_stickiness(self):
...
+                                                                                         access_log_bucket_prefix,
+                                                                                         access_log_emit_interval)
+                load_balancer_attributes['AccessLog'] = {
+                        'Enabled': True,
+                        'S3BucketName': access_log_bucket_name,
+                        'EmitInterval': access_log_emit_interval,
+                        'S3BucketPrefix': access_log_bucket_prefix
+                }
+                
+            if load_balancer_attributes:
```
Comment (sijis) [inline src/foremast/elb/create_elb.py]: I dont think the conditional is needed.
Just do the `
log.info(..)
log.debug(...)
elbclient.modify_load_balancer_attributes()`

What you can do, is earlier (when checking those custom attributes are defined), within the log, add `log.info('Setting custom ELB AccessLog path ...')`. (add 'custom' in the log output)

File: src/foremast/elb/create_elb.py [hunk 2]
```diff
@@ -208,3 +221,68 @@ def add_stickiness(self):
...
+                 'ConnectionDraining': {
+                     'Enabled': False,
+                 },
+                 'ConnectionSettings': {
+                     'IdleTimeout': 60
+                 }
+            }
+            if elb_settings.get('connection_draining_timeout'):
+                connection_draining_timeout = int(elb_settings['connection_draining_timeout'])
+                log.info("Adding Backend Connection Draining of %d", connection_draining_timeout)
```
Comment (sijis) [inline src/foremast/elb/create_elb.py]: Single quotes :)

File: src/foremast/elb/create_elb.py [hunk 3]
```diff
@@ -208,3 +221,68 @@ def add_stickiness(self):
...
+                load_balancer_attributes['ConnectionDraining'] = {
+                        'Enabled': True,
+                        'Timeout': connection_draining_timeout
+                }
+            if elb_settings.get('idle_timeout'):
+                idle_timeout = int(elb_settings['idle_timeout'])
+                log.info("Adding ELB Idle Timeout of %d", idle_timeout)
```
Comment (sijis) [inline src/foremast/elb/create_elb.py]: Single quotes :)

File: src/foremast/elb/create_elb.py [hunk 4]
```diff
@@ -208,3 +221,68 @@ def add_stickiness(self):
...
+                load_balancer_attributes['ConnectionSettings'] = {
+                        'IdleTimeout': idle_timeout
+                }
+            if elb_settings.get('access_log'):
+                access_log_bucket_name = elb_settings['access_log']['bucket_name']
+                access_log_bucket_prefix = elb_settings['access_log']['bucket_prefix']
+                access_log_emit_interval = int(elb_settings['access_log']['emit_interval'])
+                log.info("Setting ELB AccessLog path to %s/%s writing every %d minutes", access_log_bucket_name,
```
Comment (sijis) [inline src/foremast/elb/create_elb.py]: Single quotes :)

File: src/foremast/elb/create_elb.py [hunk 5]
```diff
@@ -208,3 +221,68 @@ def add_stickiness(self):
...
+            if elb_settings.get('idle_timeout'):
+                idle_timeout = int(elb_settings['idle_timeout'])
+                log.info("Adding ELB Idle Timeout of %d", idle_timeout)
+                load_balancer_attributes['ConnectionSettings'] = {
+                        'IdleTimeout': idle_timeout
+                }
+            if elb_settings.get('access_log'):
+                access_log_bucket_name = elb_settings['access_log']['bucket_name']
+                access_log_bucket_prefix = elb_settings['access_log']['bucket_prefix']
+                access_log_emit_interval = int(elb_settings['access_log']['emit_interval'])
```
Comment (sijis) [inline src/foremast/elb/create_elb.py]: I'm wondering if you can simplify this as:
`
bucket_name, bucket_prefix, emit_interval = elb_settings['access_log'].values()
`
This assumes, of course, that there are only 3 keys inside 'access_log'.

File: src/foremast/elb/create_elb.py (hunk 3 above)
Comment (jvasallo) [inline src/foremast/elb/create_elb.py]: Done!

File: src/foremast/elb/create_elb.py (hunk 2 above)
Comment (jvasallo) [inline src/foremast/elb/create_elb.py]: Done!

File: src/foremast/elb/create_elb.py (hunk 4 above)
Comment (jvasallo) [inline src/foremast/elb/create_elb.py]: Done!

File: src/foremast/elb/create_elb.py (hunk 5 above)
Comment (jvasallo) [inline src/foremast/elb/create_elb.py]: That is correct, but what is to stop someone from making the items out of order? This is explicit and no question on order.

File: src/foremast/elb/create_elb.py (hunk 1 above)
Comment (jvasallo) [inline src/foremast/elb/create_elb.py]: Done!

**Pull request merged**